core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -276,20 +282,16 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
         assertEquals(partition, recordMetadata.partition)
       }
 
-      val leader1 = leaders(partition)
+      consumer.subscribe(List(topic))
```

Comment:
Is there a reason we are using `subscribe` instead of `assign`? The latter is closer to the simple consumer.